instrumentation/armeria/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaTelemetryBuilder.java

```diff
@@ -11,98 +11,92 @@
-
-  private final HttpServerRouteBuilder<RequestContext> httpServerRouteBuilder =
-      HttpServerRoute.builder(ArmeriaHttpServerAttributesGetter.INSTANCE);
-
-  private Function<
-          SpanStatusExtractor<RequestContext, RequestLog>,
-          ? extends SpanStatusExtractor<? super RequestContext, ? super RequestLog>>
-      statusExtractorTransformer = Function.identity();
+  static {
+    ArmeriaInstrumenterBuilderUtil.setClientBuilderExtractor(
+        ArmeriaTelemetryBuilder::getClientBuilder);
+    ArmeriaInstrumenterBuilderUtil.setServerBuilderExtractor(
+        ArmeriaTelemetryBuilder::getServerBuilder);
+  }
 
   ArmeriaTelemetryBuilder(OpenTelemetry openTelemetry) {
-    this.openTelemetry = openTelemetry;
+    clientBuilder = ArmeriaInstrumenterBuilderFactory.getClientBuilder(openTelemetry);
+    serverBuilder = ArmeriaInstrumenterBuilderFactory.getServerBuilder(openTelemetry);
   }
 
+  /**
+   * Sets the status extractor for both client and server spans.
+   *
+   * @deprecated Use {@link #setClientStatusExtractor(Function)} or {@link
+   *     #setServerStatusExtractor(Function)} instead.
+   */
+  @Deprecated
+  @SuppressWarnings({"unchecked", "rawtypes"})
   @CanIgnoreReturnValue
   public ArmeriaTelemetryBuilder setStatusExtractor(
       Function<
               SpanStatusExtractor<RequestContext, RequestLog>,
               ? extends SpanStatusExtractor<? super RequestContext, ? super RequestLog>>
           statusExtractor) {
-    this.statusExtractorTransformer = statusExtractor;
+    clientBuilder.setStatusExtractor((Function) statusExtractor);
+    serverBuilder.setStatusExtractor((Function) statusExtractor);
+    return this;
+  }
+
+  /** Sets the status extractor for client spans. */
+  @CanIgnoreReturnValue
+  public ArmeriaTelemetryBuilder setClientStatusExtractor(
+      Function<
+              SpanStatusExtractor<? super ClientRequestContext, ? super RequestLog>,
+              ? extends SpanStatusExtractor<? super ClientRequestContext, ? super RequestLog>>
+          statusExtractor) {
+    clientBuilder.setStatusExtractor(statusExtractor);
+    return this;
+  }
+
+  /** Sets the status extractor for server spans. */
+  @CanIgnoreReturnValue
+  public ArmeriaTelemetryBuilder setServerStatusExtractor(
+      Function<
+              SpanStatusExtractor<? super ServiceRequestContext, ? super RequestLog>,
+              ? extends SpanStatusExtractor<? super ServiceRequestContext, ? super RequestLog>>
+          statusExtractor) {
+    serverBuilder.setStatusExtractor(statusExtractor);
     return this;
   }
 
   /**
    * Adds an additional {@link AttributesExtractor} to invoke to set attributes to instrumented
    * items. The {@link AttributesExtractor} will be executed after all default extractors.
+   *
+   * @deprecated Use {@link #addClientAttributeExtractor(AttributesExtractor)} or {@link
+   *     #addServerAttributeExtractor(AttributesExtractor)} instead.
    */
+  @Deprecated
```

Comment:
Just wanted to point out that since this types cleanly we don't really need to deprecate and remove it. @trask what is your preference. I'm fine with both keeping and removing it.

Reply:
thanks for pointing it out, I'm good deprecating / removing it